code/modules/surgery/organs/augments_internal.dm

```diff
@@ -253,6 +253,73 @@
+		return
+	if(out)//check if the owner has the item out already
+		owner.unEquip(holder, 1)//if he does, take it away. then,
+		holder.loc = null//stash it in nullspace
+		out = 0//and set this to clarify the item isn't out.
+		owner.visible_message("<span class='notice'>You retract [holder].</span>")
+		playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+	else//if he doesn't have the item out
+		if(owner.put_in_hands(holder))//put it in his hands.
+			out = 1
+			owner.visible_message("<span class='notice'>You extend [holder]!</span>")
+			playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+		else//if this fails to put the item in his hands,
+			holder.loc = null//keep it in nullspace
+			owner.visible_message("<span class='warning'>You can't extend [holder] if you can't use your hands!</span>")
+
+/obj/item/organ/internal/cyberimp/chest/arm_mod/emp_act(severity)//if the implant gets EMPed...
+	if(!owner || overloaded)//ensure that it's in an owner and that it's not already EMPed, then...
+		return
+	if(out)//check if he has the item out...
+		owner.unEquip(holder, 1)//if he does, take it away.
+		holder.loc = null
+		out = 0
+		owner.visible_message("<span class='notice'>[holder] forcibly retracts into your arm.</span>")
+	owner.visible_message("<span class='danger'>A loud bang comes from [owner]...</span>")
+	playsound(get_turf(owner), 'sound/weapons/flashbang.ogg', 100, 1)
+	owner << "<span class='warning'>You feel an explosion erupt inside you as your chest implant breaks. Is it hot in here?</span>"
+	owner.adjust_fire_stacks(20)
+	owner.IgniteMob()//ignite the owner, as well as
+	owner.say("AUUUUUUUUUUUUUUUUUUGH!!")
+	owner.adjustFireLoss(25)//severely injure him!
+	overloaded = 1//then make sure this can't happen again by breaking the implant.
+
+/obj/item/organ/internal/cyberimp/chest/arm_mod/tase//mounted, self-charging taser!
+	name = "Arm-cannon taser implant"
+	desc = "A variant of the arm cannon implant that fires electrodes and disabler shots. The cannon emerges from the subject's arms and remains in the shoulders when not in use."
+	icon_state = "armcannon_tase_implant"
+	origin_tech = "materials=5;combat=5;biotech=4;powerstorage=4"
+	organ_action_name = "Toggle Arm Cannon Taser"
+
+/obj/item/organ/internal/cyberimp/chest/arm_mod/tase/New()//when the implant is created...
+	holder = new /obj/item/weapon/gun/energy/gun/advtaser/mounted(src)//assign a brand new item to it. (in this case, a gun)
+
+/obj/item/organ/internal/cyberimp/chest/arm_mod/lase//mounted, self-charging laser!
+	name = "Arm-cannon laser implant"
+	desc = "A variant of the arm cannon implant that fires lethal laser beams. The cannon emerges from the subject's arms and remains in the shoulders when not in use."
+	icon_state = "armcannon_lase_implant"
+	origin_tech = "materials=5;combat=5;biotech=4;powerstorage=4;syndicate=5"//this is kinda nutty and i might lower it
```

Comment:
Generally syndicate tech levels are given to things only traitors can get, it's admin only so I guess it doesn't matter too much